File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLAbsUserStatus.java
package com.github.badoualy.telegram.tl.api;

import com.github.badoualy.telegram.tl.core.TLObject;

/**
 * Abstraction level for the following constructors:
 * <ul>
 * <li>{@link TLUserStatusEmpty}: userStatusEmpty#9d05049</li>
 * <li>{@link TLUserStatusLastMonth}: userStatusLastMonth#77ebc742</li>
 * <li>{@link TLUserStatusLastWeek}: userStatusLastWeek#7bf09fc</li>
 * <li>{@link TLUserStatusOffline}: userStatusOffline#8c703f</li>
 * <li>{@link TLUserStatusOnline}: userStatusOnline#edb93949</li>
 * <li>{@link TLUserStatusRecently}: userStatusRecently#e26f42f1</li>
 * </ul>
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public abstract class TLAbsUserStatus extends TLObject {

    public TLAbsUserStatus() {
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLAbsUserStatus.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_BOOLEAN
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_DOUBLE
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT64
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLBytesSerializedSize
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLStringSerializedSize
import com.github.badoualy.telegram.tl.core.TLObject

/**
 * Abstraction level for the following constructors:
 * * [userStatusEmpty#9d05049][TLUserStatusEmpty]
 * * [userStatusLastMonth#77ebc742][TLUserStatusLastMonth]
 * * [userStatusLastWeek#7bf09fc][TLUserStatusLastWeek]
 * * [userStatusOffline#8c703f][TLUserStatusOffline]
 * * [userStatusOnline#edb93949][TLUserStatusOnline]
 * * [userStatusRecently#e26f42f1][TLUserStatusRecently]
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
abstract class TLAbsUserStatus : TLObject()
